sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -261,7 +263,7 @@ class JacksonParser(
     case st: StructType =>
       val fieldConverters = st.map(_.dataType).map(makeConverter).toArray
       (parser: JsonParser) => parseJsonToken[InternalRow](parser, dataType) {
-        case START_OBJECT => convertObject(parser, st, fieldConverters)
+        case START_OBJECT => convertObject(parser, st, fieldConverters).get
```

Comment:
Is it safe to use simply `get`? Sometime, this can cause `java.util.NoSuchElementException: None.get` situation.

Reply:
`None` can be returned when a row is skipped while applying filters. That could happen for root struct only which is not the case here.

Reply:
Got it.